hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieArchivedTimeline.java

```diff
@@ -152,9 +156,13 @@ public void loadCompactionDetailsInMemory(String compactionInstantTime) {
 
   public void loadCompactionDetailsInMemory(String startTs, String endTs) {
     // load compactionPlan
-    loadInstants(new TimeRangeFilter(startTs, endTs), true, record ->
-        record.get(ACTION_TYPE_KEY).toString().equals(HoodieTimeline.COMPACTION_ACTION)
-            && HoodieInstant.State.INFLIGHT.toString().equals(record.get(ACTION_STATE).toString())
+    loadInstants(new TimeRangeFilter(startTs, endTs), true,
+        record -> {
+          // Older files don't have action state set.
+          Object action = record.get(ACTION_STATE);
+          return record.get(ACTION_TYPE_KEY).toString().equals(HoodieTimeline.COMPACTION_ACTION)
+            && (action == null || HoodieInstant.State.INFLIGHT.toString().equals(action.toString()));
```

Comment:
When action equals null, the instant state is definite to be `INFLIGHT` for old version ? Can we write ta test case?

Reply:
old version is deprecated/fixed so dont know how to write unit tests for them.

Reply:
Fine, we are good to go, I did see several issues with this in-compatibility, let's merge it first.